src/Utils.cpp

```diff
@@ -0,0 +1,76 @@
+#include <yaml-cpp/yaml.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "../include/Utils.h"
+#include "../include/Common.h"
+
+using clang::tooling::runToolOnCode, clang::tooling::FrontendActionFactory,
+    clang::tooling::CommonOptionsParser, clang::tooling::ClangTool,
+    clang::FrontendAction;
+using std::size_t, std::vector, std::string, std::cerr, std::cout, std::endl,
+    std::ifstream;
+
+std::unique_ptr<FrontendActionFactory>
+newBaseFrontendActionFactory(vector<BaseTransformation> transformations,
+                             string output_path) {
+    /* Method is similar to newFrontendActionFactory but this method
+     * supports passing arguments to the FrontendAction constructor */
+    class SimpleFrontendActionFactory : public FrontendActionFactory {
```

Comment:
Are you sure that already implemented fabrics don't support args for actions?

Reply:
Yes, there only exist two types of `newFrontendActionFactort` you can check it [here](https://clang.llvm.org/doxygen/namespaceclang_1_1tooling.html#ad3cc1c1f9c63126287ab7d13e057fb2f)